packages/devtools_app/lib/src/http/curl_command.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../primitives/utils.dart';
+import 'http_request_data.dart';
+
+class CurlCommand {
+  /// [CurlCommand] provides the ability to create a cURL command string
+  /// based on the passed [DartIOHttpRequestData].
+  ///
+  /// When [followRedirects] is false, the `--location` option is omitted from cURL.
+  /// When [multiline] is false, the command will be forced to be in a single line.
+  factory CurlCommand.from(
+    DartIOHttpRequestData data, {
+    bool followRedirects = true,
+    bool multiline = true,
+  }) {
+    return CurlCommand._(
+      commandParts: [
+        'curl',
+        if (followRedirects) '--location',
+        '--request',
+        data.method,
+        _escapeString(data.uri),
+        ..._headers(data, multiline: multiline),
+        ..._body(data, multiline: multiline),
+      ],
+    );
+  }
+
+  CurlCommand._({
+    required this.commandParts,
+  });
+
+  static const _lineBreak = '\\\n';
+
+  final List<String> commandParts;
+
+  /// Returns the cURL command as a string.
+  @override
+  String toString() {
+    return _buildCommandString(commandParts);
+  }
+
+  static List<String> _headers(
+    DartIOHttpRequestData data, {
+    required bool multiline,
+  }) {
+    final List<String> parts = [];
```

Comment:
nit: put type on RHS to match style of this codebase:
`final parts = <String>[];`